src/mono/System.Private.CoreLib/src/System/RuntimeType.Mono.cs

```diff
@@ -1793,8 +1793,15 @@ internal override FieldInfo GetField(FieldInfo fromNoninstanciated)
             else if (IsPointer)
             {
                 Type? vtype = value.GetType();
-                if (vtype == typeof(IntPtr) || vtype == typeof(UIntPtr))
+                if (vtype == typeof(IntPtr))
                     return CheckValueStatus.Success;
+
```

Comment:
Why is this needed ?

Reply:
Thanks for this. With the last commit, it is no longer necessary. Also removed from core CLR.

It was necessary previously since the `Opcodes.UnboxAny` would throw a CastClassException.